### PR TITLE
Block & report player self-interaction

### PR DESCRIPTION
**Pull request by appgurueu**
Detects & blocks interactions with yourself. Use any hacked client and punch yourself to test this.

### CODE REVIEW

**Comment by sfan5** [comment]
I'd rather put these in `src/network/serverpackethandler.cpp`
Also you might want to add a message like so
https://github.com/minetest/minetest/blob/fde2785fe363c55e8acfd17af71375a231df41c1/src/network/serverpackethandler.cpp#L974-L975

File: src/network/serverpackethandler.cpp [hunk 1]
```diff
@@ -1038,6 +1038,11 @@ void Server::handleCommand_Interact(NetworkPacket *pkt)
 		if (pointed.type == POINTEDTHING_NODE) {
 			target_pos = intToFloat(pointed.node_undersurface, BS);
 		} else if (pointed.type == POINTEDTHING_OBJECT) {
+			if (playersao->getId() == pointed_object->getId()) {
+				actionstream << player->getName() << " attempted to interact with themself" << std::endl;
```
Comment (sfan5) [inline src/network/serverpackethandler.cpp:1042]:
```suggestion
				actionstream << "Server: " << player->getName()
					<< " attempted to interact with themselves" << std::endl;
```

**Review by sfan5 — CHANGES_REQUESTED** [review]
:+1: if changed

**Comment by appgurueu** [comment]
@gnu9d ooh, your hack client ceases to work for this. cry me a river.

**Pull request merged**